bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/network/SiemensHvacConnectorImpl.java

```diff
@@ -0,0 +1,489 @@
+            request.method(HttpMethod.GET);
+
+            ContentResponse response = executeRequest(request, callback);
+            if (callback == null && response != null) {
+                int statusCode = response.getStatus();
+
+                if (statusCode == HttpStatus.OK_200) {
+                    String result = response.getContentAsString();
+
+                    return result;
+                }
+            }
+        } catch (Exception ex) {
+            logger.error("siemensHvac:DoRequest:Exception by executing Request: {} ; {} ", uri,
+                    ex.getLocalizedMessage());
+        } finally {
+        }
+
+        return null;
+    }
+
+    @Override
+    public @Nullable JsonObject DoRequest(String req, @Nullable SiemensHvacCallback callback) {
+        try {
+            String response = DoBasicRequest(req, callback);
+
+            if (response != null) {
+
+                JsonObject resultObj = getGson().fromJson(response, JsonObject.class);
+
+                if (resultObj != null && resultObj.has("Result")) {
+                    JsonObject subResultObj = resultObj.getAsJsonObject("Result");
+
+                    if (subResultObj.has("Success")) {
+                        boolean result = subResultObj.get("Success").getAsBoolean();
+                        if (result) {
+                            return resultObj;
+                        }
+                    }
+
+                }
+
+                return null;
+            }
+        } catch (Exception e) {
+            logger.error("siemensHvac:DoRequest:Exception by executing jsonRequest: {} ; {} ", req,
+                    e.getLocalizedMessage());
+        }
+
+        return null;
+    }
+
+    @Override
+    public void WaitAllPendingRequest() {
+        logger.debug("WaitAllPendingRequest:start");
+        try {
+            Thread.sleep(1000);
+            boolean allRequestDone = false;
+
+            while (!allRequestDone) {
+                int idx = 0;
+
+                allRequestDone = true;
+                while (idx < 5 && allRequestDone) {
+                    logger.debug("WaitAllPendingRequest:waitAllRequestDone {} : {} ({}/{})", idx,
+                            (startedRequest - completedRequest), startedRequest, completedRequest);
+                    if (startedRequest != completedRequest) {
+                        allRequestDone = false;
+                    }
+                    Thread.sleep(1000);
+                    idx++;
+                }
+            }
+        } catch (InterruptedException ex) {
+            logger.debug("WaitAllPendingRequest:interrupted in WaitAllRequest");
+        }
+
+        logger.debug("WaitAllPendingRequest:end WaitAllPendingRequest");
+    }
+
+    @Override
+    public void WaitNoNewRequest() {
+        logger.debug("WaitNoNewRequest:start");
+        try {
+            int lastRequest = startedRequest;
+            Thread.sleep(5000);
```

Comment:
I don't have any good suggestions at this point, so just putting comment to not forget. This seems slightly suspicious. Same for the other `sleep`'s above and below.

Reply:
I'm afraid it would need quite some effort from both of us to find a solution meeting the requirements, but without `Thread.sleep` here and the few other places. Let's leave it as is for now, and hope it works well.

Reply:
I don't see anything wrong at runtime on this part since a lot of time.
I've now running the binding for several month, and have reinitialise it a lot of time, so the code have done a lot of test cycle.

The only way two fail there is if you have a very high cpu usage, and that the binding is not able to push new request for long time (>5s), which should be ok in most case.

The only other way two go will be to be able to know how many datapoint we have two be read before we start.
But I don't see a good way to have it be done.